examples/simple-service-poc/simple_service.py

```diff
@@ -58,20 +59,27 @@ async def get_payload():
         )
 
     async def start(self):
-        # handler responsible for starting the service
+        """handler responsible for starting the service."""
+
+        # perform the initialization of the Service
+        # (which includes sending the network details within the `deploy` command)
         async for script in super().start():
             yield script
-        self._ctx.run(self.SIMPLE_SERVICE_CTL, "--start")
-        yield self._ctx.commit()
+
+        # start the service
+        s = self._ctx.new_script()
+        s.run(self.SIMPLE_SERVICE_CTL, "--start")
+        yield s
 
     async def run(self):
         # handler responsible for providing the required interactions while the service is running
         while True:
             await asyncio.sleep(10)
-            self._ctx.run(self.SIMPLE_SERVICE, "--stats")  # idx 0
-            self._ctx.run(self.SIMPLE_SERVICE, "--plot", "dist")  # idx 1
+            s = self._ctx.new_script()
+            s.run(self.SIMPLE_SERVICE, "--stats")  # idx 0
+            s.run(self.SIMPLE_SERVICE, "--plot", "dist")  # idx 1
 
-            future_results = yield self._ctx.commit()
+            future_results = yield s
             results = await future_results
             stats = results[0].stdout.strip()
             plot = results[1].stdout.strip().strip('"')
```

Comment:
Again, see my previous comments about getting results from commands individually.

Reply:
and again, "I encountered some issues with the new approach and didn't want to distract myself further yet ..." ...